apps/src/templates/teacherNavigation/ElementOrEmptyPage.tsx

```diff
@@ -27,82 +31,159 @@ const ElementOrEmptyPage: React.FC<ElementOrEmptyPageProps> = ({
+      imageComponent: (
+        <img src={TeacherDashboardEmptyState} alt={i18n.almostThere()} />
+      ),
+      button: null,
+    },
+    noCalendarForThisUnit: {
+      headline: i18n.calendarNotAvailable(),
+      descriptionText: null,
+      imageComponent: (
+        <img src={TeacherDashboardEmptyState} alt={i18n.almostThere()} />
+      ),
+      button: null,
+    },
+    noUnitAssignedForCalendarOrLessonMaterials: {
+      headline: i18n.almostThere(),
+      descriptionText: i18n.noUnitAssigned({
+        page: lessonMaterialsOrCalendarPage,
+        courseName: courseName ? courseName : i18n.thisCourse(),
+      }),
+      imageComponent: (
+        <img src={TeacherDashboardEmptyState} alt={i18n.almostThere()} />
+      ),
+      button: (
         <Button onClick={navigateToCoursePage} text={i18n.assignAUnit()} />
-      );
-    }
+      ),
+    },
   };
 
-  const heading =
-    showNoStudents || showNoCurriculumAssigned
-      ? i18n.emptySectionHeadline()
-      : i18n.almostThere();
+  // separate out logic for the calendar page
+  const calendarState = showNoCalendarForLegacyCourses
+    ? EMPTY_STATE.noCalendarForLegacyCourses
+    : showNoUnitAssigned
+    ? EMPTY_STATE.noUnitAssignedForCalendarOrLessonMaterials
+    : EMPTY_STATE.noCalendarForThisUnit;
 
-  const navigate = useNavigate();
+  let currentEmptyState;
 
-  const navigateToCoursePage = () => {
-    navigate(`../${TEACHER_NAVIGATION_PATHS.courseOverview}`, {
-      relative: 'path',
-    });
-  };
+  if (showNoStudents) {
+    currentEmptyState = EMPTY_STATE.noStudents;
+  } else if (showNoCurriculumAssigned) {
+    currentEmptyState = EMPTY_STATE.noCurriculumAssigned;
+  } else if (isOnCalendarPage) {
+    currentEmptyState = calendarState;
+  } else if (showNoLessonMaterialsForLegacyCourses) {
+    currentEmptyState = EMPTY_STATE.noLessonMaterialsForLegacyCourses;
+  } else if (showNoLessonMaterialsForLesson) {
+    currentEmptyState = EMPTY_STATE.noLessonMaterialsForThisLesson;
+  } else {
+    currentEmptyState = EMPTY_STATE.noUnitAssigned;
+  }
+  const {imageComponent, headline, descriptionText, button} = currentEmptyState;
 
   // Don't show the empty state if we're still loading data
   if (
     isLoadingSectionData ||
-    (!showNoStudents && !showNoCurriculumAssigned && !showNoUnitAssigned)
+    (!showNoStudents &&
+      !showNoCurriculumAssigned &&
+      !showNoUnitAssigned &&
+      !showNoCalendarForLegacyCourses &&
+      !showNoCalendarForThisUnit &&
+      !showNoLessonMaterialsForLegacyCourses &&
+      !showNoLessonMaterialsForLesson)
   ) {
     return element;
   } else {
     return (
-      <div className={dashboardStyles.emptyClassroomDiv}>
-        <div className={dashboardStyles.emptyClassroomDiv}>
-          {displayedImage()}
-          <Heading3 className={styles.topPadding}>{heading}</Heading3>
-          <BodyTwoText>{textDescription()}</BodyTwoText>
-          {link()}
-        </div>
-      </div>
+      <EmptyState
```

Comment:
Created a new `EmptyState` component to hold these elements.  I imagine Molly might be doing more similar `EmptyState` graphics elsewhere in the future too...